src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java

```diff
@@ -65,13 +68,72 @@
  *  default: // OK
  *    break;
  * }
- * return switch (option) { // OK, the compiler requires switch expression to be exhaustive
- *  case ONE:
- *    yield 1;
- *  case TWO:
- *    yield 2;
- *  case THREE:
- *    yield 3;
+ * switch (o) {
+ *     case String s: // type pattern
+ *         System.out.println(s);
+ *         break;
+ *     case Integer i: // type pattern
+ *         System.out.println("Integer");
+ *         break;
+ *     default:    // will not compile without default label, thanks to type pattern label usage
+ *         break;
+ * }
+ * </pre>
+ * <p>Example of correct code which does not require default labels:</p>
+ * <pre>
+ * sealed interface S permits A, B, C {}
+ * final class A implements S {}
+ * final class B implements S {}
+ * record C(int i) implements S {}  // Implicitly final
+ *
+ * /**
+ *  * The completeness of a switch statement can be
+ *  * determined by the contents of the permits clause
+ *  * of interface S. No default label or default case
+ *  * label is allowed by the compiler in this situation, so
+ *  * this check does not enforce a default label in such
+ *  * statements.
+ *  *&#47;
+ * static void showSealedCompleteness(S s) {
+ *     switch (s) {
+ *         case A a: System.out.println("A"); break;
+ *         case B b: System.out.println("B"); break;
+ *         case C c: System.out.println("C"); break;
+ *     }
+ * }
+ *
+ * /**
+ *  * A total type pattern matches all possible inputs,
+ *  * including null. A default label or
+ *  * default case is not allowed by the compiler in this
+ *  * situation. Accordingly, check does not enforce a
+ *  * default label in this case.
+ *  *&#47;
+ * static void showTotality(String s) {
+ *     switch (s) {
+ *         case Object o: // total type pattern
+ *             System.out.println("o!");
+ *     }
+ * }
+ *
+ * enum Color { RED, GREEN, BLUE }
+ *
+ * /**
+ *  * In the case of switch expressions, the compiler requires
+ *  * that they are exhaustive, so this check does not enforce
+ *  * a default label. This code would still compile with a
+ *  * default label, but it is unnecessary so this check does
+ *  * not enforce it.
```

Comment:
This absolutely same as we have default in regular switch .
To catch unexpected extending of  enum. Does complier place compilation error on such enum if enum extended for new element?

Reply:
Please update example and doc to mention fact that missed element will trigger compilation error. So Check activity is not required.

Reply:
This is shown in existing example:

```
/**
 * In the case of switch expressions, the compiler requires
 * that they are exhaustive, so this check does not enforce
 * a default label. This code would still compile with a
 * default label, but it is unnecessary so this check does
 * not enforce it.
 */

```

Reply:
It is not clear.
You are too smart on jdk syntax terminology, most users are not operating by such terminology.

Reply:
Updated examples and documentation.

Reply:
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8d658f0_2022035457/config_coding.html#MissingSwitchDefault